pkg/cloud/services/compute/instance.go

```diff
@@ -707,12 +707,12 @@ func (s *Service) deleteAttachInterface(eventObject runtime.Object, instanceID,
 
 	mc := metrics.NewMetricPrometheusContext("router_interface", "delete")
 	err = attachinterfaces.Delete(s.computeClient, instanceID, portID).ExtractErr()
-	if mc.ObserveRequest(err) != nil {
+	if err != nil {
 		if capoerrors.IsNotFound(err) {
 			return nil
```

Comment:
we also have to observe here, otherwise the request will be forgotten
```suggestion
			return mc.ObserveRequest(nil)
```